enterprise_access/apps/bffs/tests/test_context.py

```diff
@@ -0,0 +1,68 @@
+"""
+Text for the BFF context
+"""
+from django.test import RequestFactory, TestCase
+
+from enterprise_access.apps.api_client.tests.test_constants import DATE_FORMAT_ISO_8601
+from enterprise_access.apps.bffs.context import HandlerContext
+from enterprise_access.apps.core.tests.factories import UserFactory
+from enterprise_access.utils import _curr_date
+
+
+class TestHandlerContext(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.factory = RequestFactory()
+        self.mock_user = UserFactory()
+        self.mock_error_user_message = "No associated enterprise customer uuid"
+        self.mock_error_developer_message = "Unable to complete request, created: {}, X_Request_ID: {}".format(
+            _curr_date(DATE_FORMAT_ISO_8601),
+            "12345"
+        )
+
+    def test_handler_context_init(self):
+        request = self.factory.get('sample/api/call')
+        request.user = self.mock_user
+        context = HandlerContext(request)
+
+        self.assertEqual(context.request, request)
+        self.assertEqual(context.user, self.mock_user)
+        self.assertEqual(context.data, {})
+        self.assertEqual(context.errors, [])
+        self.assertEqual(context.warnings, [])
+        self.assertEqual(context.enterprise_customer_uuid, None)
+        self.assertEqual(context.lms_user_id, None)
+
+    def test_handler_context_add_error(self):
+        request = self.factory.get('sample/api/call')
+        request.user = self.mock_user
+        context = HandlerContext(request)
+        expected_output = {
+            "developer_message": "No enterprise uuid associated to the user mock-uuid",
+            "user_message": "You may not be associated with the enterprise.",
+        }
+        arguments = {
+            **expected_output,
+            "status": 403
+        }
+        context.add_error(
+            **arguments,
+        )
+        self.assertEqual(expected_output, context.errors[0])
+
+    def test_handler_context_add_warning(self):
+        request = self.factory.get('sample/api/call')
+        request.user = self.mock_user
+        context = HandlerContext(request)
+        expected_output = {
+            "developer_message": "Heuristic Expiration",
+            "user_message": "The data received might be out-dated",
+        }
+        arguments = {
+            **expected_output,
+            "status": 113
+        }
+        context.add_warning(
+            **arguments,
+        )
+        self.assertEqual(expected_output, context.errors[0])
```

Comment:
[clarification] This is asserting calling `add_warning` but is asserting on `context.errors[0]`. Should this be `context.warnings[0]`?